src/confdb/confdb.c

```diff
@@ -1310,9 +1310,15 @@ static int confdb_get_domain_internal(struct confdb_ctx *cdb,
         }
     } else {
         /* default */
-        domain->case_sensitive = true;
-        domain->case_preserve = true;
+        if (strcasecmp(domain->provider, "ad") == 0) {
```

Comment:
Yes. It may be good to revisit the ad_get_common_options if there are some duplicities with the init function, however it does some additional checks that I do not want to touch, because the test I wrote does not cover ad_get_common_options at all.